src/Orleans.Serialization.FSharp/FSharpCodecs.cs

```diff
@@ -220,8 +220,8 @@ public FSharpValueOption<T> DeepCopy(FSharpValueOption<T> input, CopyContext con
     [RegisterSerializer]
     public class FSharpChoiceCodec<T1, T2> : IFieldCodec<FSharpChoice<T1, T2>>, IDerivedTypeCodec
     {
-        private static readonly Type ElementType1 = typeof(T1);
-        private static readonly Type ElementType2 = typeof(T2);
+        private readonly Type ElementType1 = typeof(T1);
```

Comment:
Should we inline these instead?

Reply:
Inlining `typeof(T)` needs to do complex dictionary lookups (similar to static fields) in generic types with reference type parameters. An instance field access is always fast, regardless of type parameters.

Reply:
Ok, that sounds reasonable to me

Reply:
I wonder if there are places where we should lift `typeof(T)` to fields, then.